server/raft.go

```diff
@@ -1728,12 +1731,17 @@ func (n *raft) Stop() {
 	n.shutdown(false)
 }
 
+func (n *raft) WaitForStop() {
+	n.wg.Wait()
```

Comment:
Should we check that the state is set to stopped to avoid calling in here and never returning?